### PR TITLE
[Chat]Clean local blob for image upload and check messageToLong when sending/submitting

### PR DESCRIPTION
**Pull request by Leah-Xia-Microsoft**
# What

 - Clean local blob for image upload at the component level after image is removed or after the send or richTextEditor is unmounted
 - Check messageToLong when sending/submitting with the updated information from inlineImagesWithProgress. This is a temporary workaround to make sure after the upload, we check the content length with the result from inlineImagesWithProgress.
# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by palatter — APPROVED** [review]
Tested using breakpoints

File: packages/react-components/src/components/RichTextEditor/RichTextEditor.tsx [hunk 1]
```diff
@@ -215,9 +235,13 @@ export const RichTextEditor = React.forwardRef<RichTextEditorComponentRef, RichT
...
         let removedInlineImages: Record<string, string>[] = [];
         /* @conditional-compile-remove(rich-text-editor-image-upload) */
-        if (shouldUpdateInlineImages) {
+        if (shouldRemoveInlineImages) {
           /* @conditional-compile-remove(rich-text-editor-image-upload) */
           removedInlineImages = getRemovedInlineImages(content, previousInlineImages);
+          /* @conditional-compile-remove(rich-text-editor-image-upload) */
+          if (removedInlineImages.length > 0) {
+            removeLocalBlobs(inlineImageLocalBlobs, removedInlineImages);
+          }
```
Comment (palatter) [inline packages/react-components/src/components/RichTextEditor/RichTextEditor.tsx:244]: Minor: Do we need the two CC statements below if we have one for parent if?
```suggestion
        /* @conditional-compile-remove(rich-text-editor-image-upload) */
        if (shouldRemoveInlineImages) {
          removedInlineImages = getRemovedInlineImages(content, previousInlineImages);
          if (removedInlineImages.length > 0) {
            removeLocalBlobs(inlineImageLocalBlobs, removedInlineImages);
          }
```

**Review by jpeng-ms — APPROVED** [review]
approve to unblock, please ask @emlynmac to review this

File: packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMessageComponentAsRichTextEditBox.tsx [hunk 2]
```diff
@@ -164,10 +168,15 @@ export const ChatMessageComponentAsRichTextEditBox = (
   }, []);
 
   const textTooLongMessage = useMemo(() => {
-    return messageState === 'too long'
+    return messageState === 'too long' ||
```
Comment (emlynmac) [inline packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMessageComponentAsRichTextEditBox.tsx:171]: Where does this magic text come from?

File: packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMessageComponentAsRichTextEditBox.tsx [hunk 3]
```diff
@@ -200,6 +209,22 @@ export const ChatMessageComponentAsRichTextEditBox = (
...
+    if (inlineImagesWithProgress && inlineImagesWithProgress.length > 0) {
+      const contentWithUpdatedInlineImagesInfo = getContentWithUpdatedInlineImagesInfo(
+        textValue,
+        inlineImagesWithProgress
+      );
+      const messageTooLong = isMessageTooLong(contentWithUpdatedInlineImagesInfo.length);
+      // Set contentValueWithInlineImagesOverflow state to display the error bar
+      setContentValueWithInlineImagesOverflow(messageTooLong);
+      // The change from the setContentValueOverflow in the previous line will not kick in yet.
+      // We need to relay on the local value of messageTooLong to return early if the message is too long.
```
Comment (emlynmac) [inline packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMessageComponentAsRichTextEditBox.tsx:222]:
```suggestion
      // We need to rely on the local value of messageTooLong to return early if the message is too long.
```
I think this was meant to be rely?

File: packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMessageComponentAsRichTextEditBox.tsx [hunk 4]
```diff
@@ -164,10 +168,15 @@ export const ChatMessageComponentAsRichTextEditBox = (
   }, []);
 
   const textTooLongMessage = useMemo(() => {
```
Comment (emlynmac) [inline packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMessageComponentAsRichTextEditBox.tsx:170]: This is a boolean, not a message?

**Pull request merged**